Library/ViewModels/PledgeViewModel.swift

```diff
@@ -338,6 +363,69 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
     let createBackingTransactionSuccess = project.takeWhen(createBackingEventSuccess)
 
     self.goToThanks = Signal.merge(applePayTransactionCompleted, createBackingTransactionSuccess)
+
+    let amountChanged = Signal.combineLatest(
+      backing.map { $0.pledgeAmount },
+      self.pledgeAmountDataSignal.map { $0.0 }
+    )
+    .map(!=)
+
+    let shippingRuleChanged = Signal.merge(
+      self.viewDidLoadProperty.signal.mapConst(false),
+      Signal.combineLatest(
+        backing.map { $0.locationId },
+        self.shippingRuleSelectedSignal.map { $0.location.id }
+      )
+      .map(!=)
+    )
+
+    let valuesChanged = Signal.combineLatest(
+      amountChanged,
+      shippingRuleChanged
+    )
+    .map { amountChanged, shippingRuleChanged -> Bool in
+      [amountChanged, shippingRuleChanged].contains(true)
```

Comment:
Good old `contains` 😄 👍

Reply:
Lol might I suggest `amountChanged || shippingRuleChanged` 😅nothing against `contains` I just think an `OR` seems to illustrate the logic more clearly.